packages/web-components/fast-foundation/src/form-associated/form-associated.ts

```diff
@@ -660,7 +661,10 @@ export function FormAssociated<T extends ConstructableFormAssociated>(BaseCtor:
         private _keypressHandler(e: KeyboardEvent): void {
             switch (e.key) {
                 case keyEnter:
-                    if (this.form instanceof HTMLFormElement) {
+                    if (
+                        this.form instanceof HTMLFormElement &&
+                        !(this instanceof FASTButton)
+                    ) {
```

Comment:
Is it possible to add a test for this which illustrates this preventing the current behavior?